src/js/media-controller.js

```diff
@@ -238,6 +238,10 @@ class MediaController extends MediaContainer {
       },
       'ratechange': () => {
         this.propagateMediaState(MediaUIAttributes.MEDIA_PLAYBACK_RATE, this.media.playbackRate);
+      },
+      'waiting,stalled,playing': () => {
```

Comment:
sorry I think this is inconsistent with a previous round of feedback, but I just realized we have a few more events where we're checking the `readyState` in use in our player efforts. I think we should minimally add these as well:
```waiting,stalled,playing,canplay,loadedmetadata'

Reply:
I kinda went trough them and some make less sense to me.

- `playing` makes more sense than `timeupdate` because `timeupdate` will fire many times while the playhead is moving while only one event is needed to hide the spinner when playback is resumed.
- `canplay` fires in a dubious state,  the video might not yet be playing so it would be strange to hide the loading indicator here.
- `loadedmetadata` if this comes way before waiting or stalled and the player is playing yes makes sense then

--- 

related to `timeupdate` and efficiency should media-chrome add a gate to setAttribute if the attribute did not change?
https://github.com/muxinc/media-chrome/blob/f476f82d55dfbd22bc2ee0fcb0cba56107ffca56/src/js/media-controller.js#L508-L520

Reply:
> related to timeupdate and efficiency should media-chrome add a gate to setAttribute if the attribute did not change?

I **_thought_** `attributeChangedCallback` only got invoked if, in fact, that attribute **_changed_**. If that's not the case, I think a gate would be 👌 (less worried about extra `removeAttribute` method calls, though maybe I shouldn't be?)

Reply:
> * `canplay` fires in a dubious state,  the video might not yet be playing so it would be strange to hide the loading indicator here.
> * `loadedmetadata` if this comes way before waiting or stalled and the player is playing yes makes sense then

I specifically suggested adding these because they are directly related to changes in `media.readyState`.

That said, we can always issue a followup PR if you have sufficient concern with adding them. I'll pre-approve and defer to your discretion on whether or not you add any/all of them.

Reply:
> 

thanks, true let's see if we run into issues where the state is not updating correctly.

double checked to be sure, `attributeChangedCallback` always fires if an attribute is set
https://codesandbox.io/s/strange-golick-c10fz?file=/src/index.js

Reply:
`stalled` actually shouldn't be here. I know it sounds like it should but it's related to networkState, not readyState. The _network_ can be stalling when the video is still playing. Unfortunate naming choices.